### PR TITLE
doc: release-notes-3.7: Add notes for selected network components

### PR DESCRIPTION
**Pull request by jukkar**
Added release notes for these network components:

* DNS/mDNS/LLMNR
* gPTP/PTP
* Sockets
* Capture API
* Virtual interface
* VLAN
* Wi-Fi

### CODE REVIEW

**Comment by aescolar** [comment]
Related to #75103

File: doc/releases/release-notes-3.7.rst [hunk 1]
```diff
@@ -519,21 +519,10 @@ Drivers and Sensors
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:519]: As you need to do a new push anyhow, could you also fix this:
https://github.com/zephyrproject-rtos/zephyr/pull/75103#discussion_r1657410483

**Review by aescolar — COMMENTED** [review]
Apart from the typo detected by the check it looks fine to me

**Review by rlubos — APPROVED** [review]
LGTM

**Review by aescolar — APPROVED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -596,6 +585,25 @@ Networking
...
+  * Added support for caching DNS records.
+  * Fixed error codes returned when socket creation fails, and when all results have been returned.
+  * Fixed DNS retransmission timeout calculation.
+
+* gPTP/PTP:
+
+  * Added support for IEEE 1588-2019 PTP.
+  * Added support for SO_TIMESTAMPING socket option to get timestamping information in socket
+    ancillary data.
+  * Fixed race condition on timestamp callback.
```
Comment (awojasinski) [inline doc/releases/release-notes-3.7.rst:604]: If I get it right this refers to the https://github.com/zephyrproject-rtos/zephyr/commit/c87c36d3130b60451d99ce2dec17ad42788e5d5d. The issue was that `net_pkt` was prematurely unreferenced. And accessing it from timestamp callback might resulted in undefined behaviour - it could be already reused for other packet. I would rephrase it to "Fixed packet reference count upon queuing for TX timestamping, resolving data access issues in timestamp callbacks." but either way LGTM

**Review by awojasinski — APPROVED** [review]
